### PR TITLE
Dynamically set hostnames

### PR DESCRIPTION
**Pull request by deads2k**
proof of https://github.com/openshift/library-go/pull/270

/hold

illegal fake bump

/assign @sttts

### CODE REVIEW

File: pkg/operator/certrotationcontroller/dynamic_serving.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package certrotationcontroller
+
+import (
+	"sync"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type DynamicServingRotation struct {
```
Comment (sttts) [inline pkg/operator/certrotationcontroller/dynamic_serving.go]: what is this? comments are missing

**Comment by deads2k** [comment]
comments added

File: vendor/github.com/openshift/library-go/pkg/operator/certrotation/client_cert_rotation_controller.go [hunk 2]
```diff
@@ -158,6 +160,25 @@ func (c *CertRotationController) Run(workers int, stopCh <-chan struct{}) {
 
 	}, time.Minute, stopCh)
 
+	// if we have a need to force rechecking the cert, use this channel to do it.
+	var targetRefresh <-chan struct{}
+	if refresher, ok := c.TargetRotation.CertCreator.(TargetCertRechecker); ok {
+		targetRefresh = refresher.RecheckChannel()
+	}
+	if targetRefresh != nil {
```
Comment (sttts) [inline vendor/github.com/openshift/library-go/pkg/operator/certrotation/client_cert_rotation_controller.go]: move this into the first if clause.

**Comment by deads2k** [comment]
Spoke on slack.   We're agreed this is the correct direction.

**Pull request merged**